pandas/io/common.py

```diff
@@ -713,15 +714,23 @@ def get_handle(
     # Convert BytesIO or file objects passed with an encoding
     is_wrapped = False
     if is_text and (compression or _is_binary_mode(handle, ioargs.mode)):
-        handle = TextIOWrapper(
-            # error: Argument 1 to "TextIOWrapper" has incompatible type
-            # "Union[IO[bytes], IO[Any], RawIOBase, BufferedIOBase, TextIOBase, mmap]";
-            # expected "IO[bytes]"
-            handle,  # type: ignore[arg-type]
-            encoding=ioargs.encoding,
-            errors=errors,
-            newline="",
-        )
+        # Use our custom BytesIOWrapper
+        # when reading text to bytes, otherwise use TextIOWrapper
+        if ioargs.mode == "rb" and isinstance(handle, TextIOBase):
```

Comment:
I think it would be clearer to move this out of the `if is_text` block (and put it in a `if not is_text` block) as the returned file handle returns/expects bytes?

Reply:
ok, I refactored it out of the is_text block. Now, it just triggers when you are trying to read bytes ("rb") from a string buffer.